cli/cmdlineparser.cpp

```diff
@@ -1653,6 +1638,30 @@ CmdLineParser::Result CmdLineParser::parseFromArgs(int argc, const char* const a
         }
     }
 
+    if (!platform.empty())
+    {
+        std::string errstr;
+        if (!mSettings.platform.set(platform, errstr, lookupPaths, mSettings.debuglookup || mSettings.debuglookupPlatform)) {
+            mLogger.printError(errstr);
+            return Result::Fail;
+        }
+
+        // TODO: remove
+        // these are loaded via external files and thus have Settings::PlatformFile set instead.
+        // override the type so they behave like the regular platforms.
+        if (platform == "unix32-unsigned") {
+            mSettings.platform.type = Platform::Type::Unix32;
+            mLogger.printMessage("The platform 'unix32-unsigned' has been deprecated and will be removed in Cppcheck 2.19. Please use '--platform=unix32 --funsigned-char' instead");
+        }
+        else if (platform == "unix64-unsigned") {
+            mSettings.platform.type = Platform::Type::Unix64;
+            mLogger.printMessage("The platform 'unix64-unsigned' has been deprecated and will be removed in Cppcheck 2.19. Please use '--platform=unix64 --funsigned-char' instead");
+        }
+    }
+
+    if (defaultSign != '\0')
+        mSettings.platform.defaultSign = defaultSign;
```

Comment:
This does not reflect the left-to-right processing as the imported project might change this(?). Even more reason for #7293.

Also make me wonder if this flag could be set per file and thus needs to life inside `FileSettings` instead of the platform after all...

Reply:
> Also make me wonder if this flag could be set per file and thus needs to life inside FileSettings instead of the platform after all...

yeah. that seems to be a edge case to me. I am not 100% sure if we need to support analysing multi-platform builds in 1 analysis.

Reply:
> yeah. that seems to be a edge case to me. I am not 100% sure if we need to support analysing multi-platform builds in 1 analysis.

That is the misconception. The signedness of char has *nothing* to do with the platform. It is dependent on the compiler and can be controlled individually with a CLI option. It is like specifying the language or standard.

Reply:
We already set the platform type per file so this would be akin to that.

Reply:
Either way it is out of the scope of this PR.